src/arrayMethodSort.js

```diff
@@ -5,8 +5,26 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
```

Comment:
```suggestion
  [].__proto__.sort2 = function(compareFunction = defaultCompareFunction) {
```
You can create a default sorting function and set it as a default value of compareFunction.
It will help to avoid extra checks and make your code cleaner